src/main/java/sign_up/drivers/SignUpDatabaseAccess.java

```diff
@@ -0,0 +1,191 @@
+package sign_up.drivers;
+
+import com.opencsv.CSVWriter;
+import com.opencsv.exceptions.CsvValidationException;
+import entities.User;
+import sign_up.use_case.SignUpDSGateway;
+import com.opencsv.CSVReader;
+
+import java.io.FileWriter;
+import java.io.Reader;
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Arrays;
+
+public class SignUpDatabaseAccess implements SignUpDSGateway {
+    private final String userFilePath;
+    private final String adminFilePath;
+    private final String numUsersCreatedFilePath;
+
+    public SignUpDatabaseAccess(String userFilePath, String adminFilePath, String numUsersCreated) {
+        this.userFilePath = userFilePath;
+        this.adminFilePath = adminFilePath;
+        this.numUsersCreatedFilePath = numUsersCreated;
+    }
+
+    @Override
+    public int getNumberUsers() {
+        ArrayList<String[]> numUsers;
+        try {
+            numUsers = readLineByLine(numUsersCreatedFilePath);
+        } catch(Exception e) {
+            System.out.println("Wrong Path");
+            String[] defaultEntry = {"0"};
+            numUsers = (ArrayList<String[]>) Arrays.asList(new String[][]{defaultEntry});
+        }
+
+        return Integer.parseInt(numUsers.get(1)[0]);
```

Comment:
I did a quick search online and could not figure out what the parseint method does. can you explain this to me? sorry for the hassle